### DISCUSSION TITLE
Skillsets for how to master the horizontal architecture ?

### DISCUSSION THREAD

**Discussion by bbartling** [VOLTTRON/volttron#2970] [General]
Would anyone ever be interested in a VOLTTRON office hour series on how to implement the horizontal architecture defined by Nexus Labs? I.E., _**a bootcamp for the beginner**_.

https://www.nexuslabs.online/part-3-the-horizontal-architecture/

What I am hoping to understand hopefully before retirement:

- How does the meta data get applied (whatever schema is used) via data base code or Python agent coding?
- What data bases work well? Do I need to be a database expert to do this or would VOLTTRON make things easier?
- This would require the "API" on the VOLTTRON central instance, right?
- What would be the key components in this architecture for Independent data layer, network, and device layer?
- What application layer apps would be compatible with this architecture?
- Can long term data be pulled from the "API"? I.E., if a fault detection app requires lots of "historical" data, can the API do that? Or is the API just for instantaneous values from buildings sensors or HVAC operational data?

![image](https://user-images.githubusercontent.com/29737117/169595473-3d941761-56af-44bd-b1a4-157d8941ed69.png)

**Comment by ghrist** [comment]
Sure, why not?

**Comment by bbartling** [comment]
I know there has been quite a few office hours on the BACnet discovery processes but its still a hard one for people getting started.

I think it would be cool having a series like an office hours dedicated to each layer in the Nexus Labs' architecture. And what would be really cool is if people could code alongside, like all the bootcamps out there these days. For example:

Office hours 1) - Device/Network Layer

- VOLTTRON deployment planning/architecture
- Introduction to buildings systems (silo's) and protocols for the edge device
- Clone the repo to get started, run the bootstrap scripts, etc.
- Scan for BACnet devices on a fake BACnet device running inside some container
- What is the platform.driver?
- Start VOLTTRON edge device
- Troubleshooting

Office hours 2) - Device/Network Layer
- What is VOLTTRON Central?
- Authorization for VOLTTRON Central/edge device
- Debugging Edge Device log files


Office hours 3) - Independent Data Layer (prerequisite is SQL or database)
- Introduction to meta data
- SQLhistorian on the VOLTTRON Central instance
- New VOLTTRON API features

Office hours 4) - Independent Data Layer
- VOLTTRON Central API authentication/authorization
- Tagged based API (retrieve data via meta data)

Office hours 5) - Application Layer
- Advanced VOLTTRON created apps for fault detection and visualization
- Other 3rd party apps for visualization
- Other 3rd party apps for FDD

Office hours 6) - Application Layer

**Reply by bonicim** [reply]
> 

Hi @bbartling ,

In addition to our [ReadTheDocs](https://volttron.readthedocs.io/en/develop/index.html#), Volttron has a [For Developers web page](https://volttron.org/developers) that provides resources for folks new to Volttron. On some helpful videos that developers can use to code along with, consider out [Tutorial Platform Install](https://volttron.org/videos/volttron%E2%84%A2-tutorial-platform-install), [BACnet Tutorial](https://volttron.org/videos/bacnet-tutorial), and [Platform Install](https://volttron.org/videos/platform-install). Moreover, we have a collection of Office Hour videos for reference at https://volttron.org/videos. Some of those office hours videos have 'tutorial' in its title, which can be used to search for helpful tutorial videos.
